src/Illuminate/Cookie/CookieJar.php

```diff
@@ -138,18 +146,31 @@ public function queue(...$parameters)
             $cookie = call_user_func_array([$this, 'make'], $parameters);
         }
 
-        $this->queued[$cookie->getName()] = $cookie;
+        if (! isset($this->queued[$cookie->getName()])) {
+            $this->queued[$cookie->getName()] = [];
+        }
+
+        $this->queued[$cookie->getName()][$cookie->getPath()] = $cookie;
     }
 
     /**
      * Remove a cookie from the queue.
      *
      * @param  string  $name
+     * @param  string|null $path
      * @return void
      */
-    public function unqueue($name)
+    public function unqueue($name, $path = null)
```

Comment:
The signature should also be changed then on the interface as now they are not in sync.

Reply:
Damn, I totally missed that.
Should I add the parameter to the interface and make an upgrade harder for people who implement the interface or revert the changes for `unqueue`?